src/components/Common/UI/Button/Button.tsx

```diff
@@ -0,0 +1,41 @@
+import { Button as AriaButton } from 'react-aria-components'
+import { type ButtonProps } from './ButtonTypes'
+import styles from './Button.module.scss'
+
+export function Button({
+  variant = 'primary',
+  isError = false,
+  isLoading = false,
+  isDisabled = false,
+  buttonRef,
```

Comment:
Would it be preferable to just expose this as `ref` rather than `buttonRef`? `inputRef` makes sense for the field components since they render the input along with the helper/error text and label so it's less clear what you get the ref to. In this case tho feels like it could just be `ref`?

I could see buttonRef being more straightforward for the adapter case where we want it explicitly noted for the consumer what they need to wire up. But it'll also put us into a naming situation where we'll constantly need to prefix `ref` for our components (which may not be a big deal)